crates/core/src/config/secret.rs

```diff
@@ -0,0 +1,259 @@
+        let pk = self.transport_keypair.secret();
+        let mut key_file = File::create(&self.transport_keypair_path)?;
+        key_file.write_all(
+            pk.to_bytes()
+                .map_err(|e| {
+                    std::io::Error::new(
+                        std::io::ErrorKind::InvalidData,
+                        format!("Failed to write keypair file: {e}"),
+                    )
+                })?
+                .as_slice(),
+        )?;
+
+        let mut nonce_file = File::create(&self.nonce_path)?;
+        nonce_file.write_all(&self.nonce)?;
+
+        let mut cipher_file = File::create(&self.cipher_path)?;
+        cipher_file.write_all(&self.cipher)?;
+
+        Ok(())
+    }
+}
+
+fn read_nonce(path_to_nonce: impl AsRef<Path>) -> std::io::Result<[u8; NONCE_SIZE]> {
+    let path_to_nonce = path_to_nonce.as_ref();
+    let mut nonce_file = File::open(path_to_nonce).map_err(|e| {
+        std::io::Error::new(
+            e.kind(),
+            format!("Failed to open key file {}: {e}", path_to_nonce.display()),
+        )
+    })?;
+    let mut buf = [0u8; NONCE_SIZE];
+    nonce_file.read_exact(&mut buf).map_err(|e| {
+        std::io::Error::new(
+            e.kind(),
+            format!("Failed to read key file {}: {e}", path_to_nonce.display()),
+        )
+    })?;
+
+    Ok::<_, std::io::Error>(buf)
+}
+
+fn read_cipher(path_to_cipher: impl AsRef<Path>) -> std::io::Result<[u8; CIPHER_SIZE]> {
+    let path_to_cipher = path_to_cipher.as_ref();
+    let mut cipher_file = File::open(path_to_cipher).map_err(|e| {
+        std::io::Error::new(
+            e.kind(),
+            format!("Failed to open key file {}: {e}", path_to_cipher.display()),
+        )
+    })?;
+    let mut buf = [0u8; CIPHER_SIZE];
+    cipher_file.read_exact(&mut buf).map_err(|e| {
+        std::io::Error::new(
+            e.kind(),
+            format!("Failed to read key file {}: {e}", path_to_cipher.display()),
+        )
+    })?;
+
+    Ok::<_, std::io::Error>(buf)
+}
+
+fn read_transport_keypair(path_to_key: impl AsRef<Path>) -> std::io::Result<TransportKeypair> {
+    let path_to_key = path_to_key.as_ref();
+    let mut key_file = File::open(path_to_key).map_err(|e| {
+        std::io::Error::new(
+            e.kind(),
+            format!("Failed to open key file {}: {e}", path_to_key.display()),
+        )
+    })?;
+    let mut buf = String::new();
+    key_file.read_to_string(&mut buf).map_err(|e| {
+        std::io::Error::new(
+            e.kind(),
+            format!("Failed to read key file {}: {e}", path_to_key.display()),
+        )
+    })?;
+
+    let pk = rsa::RsaPrivateKey::from_pkcs1_pem(&buf).map_err(|e| {
+        std::io::Error::new(
+            std::io::ErrorKind::InvalidData,
+            format!("Failed to read key file {}: {e}", path_to_key.display()),
+        )
+    })?;
+
+    Ok::<_, std::io::Error>(TransportKeypair::from_private_key(pk))
+}
```

Comment:
add a unit test for loading the from disk or using default ones to see it works correctly. you can create the necessary files on a temp dir etc.